bin/resin-preload

```diff
@@ -203,33 +202,16 @@ getResinSdk(options)
       reject(err)
     })
 
-    return preloader.build()
-    .then(() => {
-      return preloader.prepare()
-    })
-    .then(() => {
-      return preloader.fetchApplication()
-    })
+    return preloader.prepare()
     .then(() => {
-      if (preloader.application.build.length === 0) {
-        reject(new ExitError('This application has no successful builds'))
-      }
-      return preloader.getDeviceTypeAndPreloadedBuilds()
-    })
-    .then((info) => {
-      if (options.dontCheckDeviceType === false && info.device_type !== preloader.application.device_type) {
-        const msg = `Application device type (${preloader.application.device_type}) and image device type (${info.device_type}) do not match.`
+      const msg = preloader.cantPreload()
+      if (msg) {
         reject(new ExitError(msg))
+      } else {
+	return preloader.preload()
       }
-      const builds = info.preloaded_builds.map((build) => {
-        return build.slice(-preload.BUILD_HASH_LENGTH)
-      })
-      if (builds.indexOf(options.commit || preloader.application.commit) !== -1) {
-        reject(new ExitError('This build is already preloaded in this image.'))
-      }
-      return preloader.preload()
-      .then(resolve)
     })
+    .then(resolve)
```

Comment:
You can resolve the promise earlier and remove these two lines (`then` and `catch` below`).

Reply:
I don't think I can do that, this code is inside a `new Promise((resolve, reject) => {...`.